packages/integration-karma/test/light-dom/basic/index.spec.js

```diff
@@ -0,0 +1,40 @@
+import { createElement, setFeatureFlagForTest, LightningElement } from 'lwc';
+
+import Test from 'x/test';
+
+describe('Basic Light DOM', () => {
+    it('should render properly', () => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+        const elm = createElement('x-test', { is: Test });
+        document.body.appendChild(elm);
+
+        expect(elm.shadowRoot).toBeNull();
+        expect(elm.firstChild.innerText).toEqual('Hello, Light DOM');
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+    it('should render to Shadow DOM when feature is disabled', () => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+        expect(() => createElement('x-test', { is: Test })).toThrowError();
+    });
+
+    it('should return null for template', () => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+
+        const spy = spyOn(console, 'error');
+        let template;
+        class TemplateTest extends LightningElement {
+            static shadow = false;
+            connectedCallback() {
+                template = this.template;
+            }
+        }
+
+        const elm = createElement('x-test', { is: TemplateTest });
+        document.body.appendChild(elm);
+        expect(template).toBeNull();
+        expect(spy.calls.mostRecent().args[0].message).toEqual(
```

Comment:
You could use the `toLogErrorDev` custom matcher for this. You can find more details about it in the `CONTRIBUTING.md` instead of using a custom spy.